icu4j/main/core/src/test/java/com/ibm/icu/dev/test/lang/UnicodeSetTest.java

```diff
@@ -3198,4 +3205,178 @@ public void TestPatternCodePointComplement() {
             assertFalse("[:Basic_Emoji:].complement() --> no bicycle", notBasic.contains("🚲"));
         }
     }
+
+    private static UnicodeSet populateUnicodeSet() {
+        // Trying to cover the most interesting combinations:
+        final UnicodeSet unicodeSet = new UnicodeSet();
+        // Patterns
+        unicodeSet.applyPattern("\\p{sc=Ethi}");
+        unicodeSet.applyPattern("\\p{Number}");
+        // Single code point
+        unicodeSet.add('X'); // adds a code point in the ASCII range
+        unicodeSet.add('Σ'); // adds a code point in the Greek block
+        unicodeSet.add(0x1F600); // adds a code point above BMP (😀)
+        // Code point ranges and from CharSequence
+        unicodeSet.add('A', 'F'); // adds a code point range in the ASCII range
+        unicodeSet.add('α', 'ζ'); // adds a code point range in the Greek block
+        unicodeSet.add(0x1F347, 0x1F353); // adds a code point range above BMP (🍇-🍓)
+        unicodeSet.addAll("hello"); // adds code points from CharSequence
+        // Strings
+        unicodeSet.add("world"); // adds string
+        unicodeSet.addAll("one", "two", "three"); // adds strings
+        return (UnicodeSet) unicodeSet.freeze();
+    }
+
+    @Test
+    public void testIterationMethodsCodepoints() {
+        for (UnicodeSet us : UNICODE_SETS_TO_ITERATE) {
+
+            // Build the reference test result
+            StringJoiner expected = newResult();
+            int i = 0;
+            while (true) {
+                int cp = us.charAt(i++);
```

Comment:
ok for test code; although (optional) you could consider using UnicodeSetIterator as the reference implementation, rather than testing it again (since it has its own test code)

----
alternative, double-loop-iterate over the ranges and then their code points?